msg-socket/src/req/driver.rs

```diff
@@ -97,6 +98,18 @@ impl<T: AsyncRead + AsyncWrite + Unpin> Future for ReqDriver<T> {
                 Poll::Pending => {}
             }
 
+            // Check for request timeouts
+            let now = std::time::Instant::now();
+            this.pending_requests.retain(|_, request| {
+                if now.duration_since(request.start) > this.options.timeout {
+                let sender_clone = mem::replace(&mut request.sender, oneshot::channel().0);
```

Comment:
Why do we need the replace? Can just send an error on the `request.sender` right?